pkg/registry/rbac/reconciliation/reconcile_role.go

```diff
@@ -194,7 +194,10 @@ func computeReconciledRole(existing, expected RuleOwner, removeExtraPermissions
 	}
 
 	// Compute extra and missing rules
-	_, result.ExtraRules = validation.Covers(expected.GetRules(), existing.GetRules())
+	// Don't compute extra permissions if expected and existing roles are both aggregated
+	if expected.GetAggregationRule() == nil || existing.GetAggregationRule() == nil {
```

Comment:
Not overly familiar with the code, but is this actually a behaviour change or just something that was harmless before but now causes extra output?

Reply:
The rules in aggregated roles are set by the aggregating controller, based on the selector in the role. If both the expected and existing roles are aggregated roles, it doesn't make sense to compute "extra" rules by looking at the expected role's rules.

reconciliation has two callers: kube-apiserver on startup, and `kubectl auth reconcile`
* kube-apiserver doesn't remove extra rules or subjects, so this is not a behavior change for it
* kubectl would have removed these extra rules if `--remove-extra-rules` was passed, then the rules would have been immediately/eventually reset by the aggregated role controller. I'd actually consider removing them a bug for an aggregated role.